lib/irq.c

```diff
@@ -376,41 +376,34 @@ handle_device_set_irqs(vfu_ctx_t *vfu_ctx, uint32_t size,
     return dev_set_irqs(vfu_ctx, irq_set, data);
 }
 
-static int
+static bool
 validate_irq_subindex(vfu_ctx_t *vfu_ctx, uint32_t subindex)
 {
     if (vfu_ctx == NULL) {
-        errno = EINVAL;
-        return -1;
+        return false;
     }
 
     if ((subindex >= vfu_ctx->irqs->max_ivs)) {
         vfu_log(vfu_ctx, LOG_ERR, "bad IRQ %d, max=%d\n", subindex,
                vfu_ctx->irqs->max_ivs);
-        /* FIXME should return -errno */
-        errno = EINVAL;
-        return -1;
+        return false;
     }
 
-    return 0;
+    return true;
 }
 
 int
 vfu_irq_trigger(vfu_ctx_t *vfu_ctx, uint32_t subindex)
 {
-    int ret;
     eventfd_t val = 1;
 
-    ret = validate_irq_subindex(vfu_ctx, subindex);
-    if (ret < 0) {
-        return ret;
+    if (validate_irq_subindex(vfu_ctx, subindex) == false) {
```

Comment:
nit, sorry, but don't ever do "== <boolean>", this should just be:

  if (!validate_irq_subindex()) {

Reply:
Yes I also do not like doing "==", but its the frodo coding_style which says so at-least for checking NULL.
Changed it to !